src/main/resources/org/nest/sympy/misc.py

```diff
@@ -0,0 +1,178 @@
+
+        if simplify(diff_rhs_lhs) == sympify(0):
+            found_ode = True
+        
+        # Initialize the (potential) order of the differential equation.
+        order = 1
+        
+        # while an ODE has not yet been found and we have not yet
+        # reached the maximum order we are checking for, we check if
+        # `shape` satisfies a linear homogeneous ODE of the next higher
+        # order.
+        while not found_ode and order < MAX_ORDER:
+
+            # The potential order must be at least `order+1`
+            order += 1
+            # Add the next higher derivative to the list of
+            # derivatives of `shape`
+            derivatives.append(diff(derivatives[-1], t))
+            
+            # The goal here is to calculate the factors (which we call
+            # `derivative_factors`) of the ODE (assuming they
+            # exist). The idea is to create a system of equations by
+            # substituting (natural) numbers into the homogeneous
+            # linear homogeneous ODE with variable `derivative_factors`
+            # order many times for varying (natural) numbers and solving for
+            # `derivative_factors`. Once we have `derivative_factors` the
+            # ODE is uniquely defined. This is assuming that shape
+            # satisfies an ODE of this order. (This we must stil
+            # check)
+        
+            # `X` will contain as rows derivatives up to `order-1` of some
+            # natural numbers (differing in each row)
+            X = zeros(order)
+            
+            # `Y` will contain the derivatives of `order` of the natural
+            # number in the corresponding row of `X`
+            Y = zeros(order, 1)
+            
+            # It is possible that by choosing certain natural numbers,
+            # the system of equations will not be solvable, i.e. `X` is
+            # not invertable. This is unlikely but we check for
+            # invertability of `X` for varying sets of natural numbers.
+            invertible = False
+            for k in range(MAX_TRIES):
+                for i in range(order):
+                    substitute = i+k+1
+                    Y[i] = derivatives[order].subs(t, substitute)
+                    for j in range(order):
+                        X[i, j] = derivatives[j].subs(t, substitute)
+                d = det(X)
+                if d != 0:
+                    invertible = True
+                    break
+
+            if not invertible:
+                raise Exception("Failed to find homogeneous linear ODE "
+                                "which shape satisfies, or shape does "
+                                "not satisfy any such ODE of order <= {}".format(MAX_ORDER))
+            
+            derivative_factors = X.inv() * Y
+            diff_rhs_lhs = 0
+            # Once we have 'derivative_factors' the
+         
+            # We calculated derivative_factors of the linear
```

Comment:
This should come before the previous explanation:

> If $f^{(order)}(t) = \sum_{i<order} C_i f^{(i)}(t)$, then we can find the
> $order-1$ derivative factors by evaluating the previous equation as a
> linear system of order $order-1$ such that Y = [f^{(order)}] = mat_X * [C_i]
> hence [C_i] can be found by inverting mat_X.

Or something like that (but the ipynb would be best I think)